meta/store.go

```diff
@@ -1664,6 +1678,27 @@ func (fsm *storeFSM) Apply(l *raft.Log) interface{} {
 	return err
 }
 
+func (fsm *storeFSM) applyRemovePeerCommand(cmd *internal.Command) interface{} {
+	ext, _ := proto.GetExtension(cmd, internal.E_RemovePeerCommand_Command)
+	v := ext.(*internal.RemovePeerCommand)
+
+	id := v.GetID()
+	addr := v.GetAddr()
+	//Remove that node from the peer
+	if id == fsm.id {
+		// if we are part of the raft cluster, remove us
+		if _, ok := fsm.raftState.(*localRaft); ok {
```

Comment:
This casting isn't needed.  A `remoteRaft` instance can never apply any log messages because it doesn't have a raft instance.